lib/resty/kafka/sendbuffer.lua

```diff
@@ -17,12 +19,14 @@ local MAX_REUSE = 10000
 local _M = {}
 local mt = { __index = _M }
 
-function _M.new(self, batch_num, batch_size)
+function _M.new(self, batch_num, batch_size, max_retry, retry_backoff)
     local sendbuffer = {
         topics = {},
         queue_num = 0,
         batch_num = batch_num * 2,
         batch_size = batch_size,
+        max_retry = max_retry or 3,
```

Comment:
There are a lot of `max_retry or 3` invocations. Consider moving this to a more central place.
Same goes for `retry_backoff`

Reply:
This retry here (`sendbuffer.lua`) is to retry the Kafka cluster nodes, while the retry in `producer.lua` is for retry to send message. They are for different purposes, but the value are both from Kong's config.

In `sendbuffer.lua` we cannot get the `retry` value from `producer.lua`.